packages/dds/tree/src/simple-tree/api/schemaFactoryAlpha.ts

```diff
@@ -51,6 +53,27 @@ export class SchemaFactoryAlpha<
 		) as ScopedSchemaName<TScope, Name>;
 	}
 
+	/**
+	 * Declares a type enablable in a set of {@link AllowedTypes}.
+	 * This means that the type can be read from the document but cannot be written.
+	 *
+	 * @remarks
+	 * t is frozen and should not be modified after being passed in.
+	 */
+	public enablable<const T extends TreeNodeSchema>(
+		t: T | AnnotatedAllowedType<T>,
+	): AnnotatedAllowedType<T> {
+		Object.freeze(t);
+		const annotatedType = normalizeToAnnotatedAllowedType(t);
```

Comment:
I think any code which normalizes a type probably will have issues if its modified after that, so putting the freeze inside normalizeToAnnotatedAllowedType would make more sense